### PR TITLE
Periphmraa

### PR DESCRIPTION
**Pull request by vintummala**
Peripheralman: Provide android things support to the Master by merging the periphmraa branch.

### CODE REVIEW

File: api/mraa/common.h [hunk 1]
```diff
@@ -234,6 +234,42 @@ unsigned int mraa_get_platform_pin_count(uint8_t platform_offset);
 char* mraa_get_pin_name(int pin);
 
 /**
+* Get pin number, board must be initialised.
+*
+* @param pin_name: GPIO Pin Name. Eg: IO0
+* @return int of MRAA index for gpio
```
Comment (pylbert) [inline api/mraa/common.h]: Is there a not found case?

Perhaps:
@return GPIO index for pin_name or -1 if not found

Applies to all mraa_*_lookup methods

File: api/mraa/common.hpp [hunk 2]
```diff
@@ -215,6 +217,90 @@ getPinName(int pin)
 }
 
 /**
+* Get pin number, board must be initialised.
```
Comment (pylbert) [inline api/mraa/common.hpp]: Same question as C methods - pins or indexes?

File: api/mraa/common.hpp [hunk 3]
```diff
@@ -215,6 +217,90 @@ getPinName(int pin)
...
+*
+* @param pin_name: GPIO Pin Name. Eg: IO0
+* @throws std::invalid_argument if name is not valid
```
Comment (pylbert) [inline api/mraa/common.hpp]: Maybe:
@throws std::invalid_argument if name is not found

File: docs/building.md [hunk 4]
```diff
@@ -184,3 +184,21 @@ build machine.
 ~~~~~~~~~~~~~{.sh}
 cmake -DRPM=ON -DCMAKE_INSTALL_PREFIX=/usr ..
 ~~~~~~~~~~~~~
+
+## Building for Peripheralmanager Android Things
+
+Change src/CMakeLists.txt:140 to the location of libandroidthings.so on your
```
Comment (pylbert) [inline docs/building.md]: The hard-coded path to a library on brendan's machine was switched out in favor of using CMake.  Assuming the github repo below is where Android will release the definition/implementation of libandroidthings then:

Requirements: Android Things Native Library (https://github.com/androidthings/native-libandroidthings)
The Android Things Native Library contains a CMake find_package module (FindAndroidThings.cmake).  Make sure the directory containing this module is added to the CMAKE_MODULE_PATH.  Example cmake -DCMAKE_MODULE_PATH=/path/to/native-libandroidthings

File: api/mraa/common.h [hunk 5]
```diff
@@ -234,6 +234,42 @@ unsigned int mraa_get_platform_pin_count(uint8_t platform_offset);
 char* mraa_get_pin_name(int pin);
 
 /**
+* Get pin number, board must be initialised.
```
Comment (pylbert) [inline api/mraa/common.h]: Are these methods returning pin numbers or indexes?

Perhaps:
Get GPIO index by name, board must be initialised

File: api/mraa/common.h [hunk 6]
```diff
@@ -234,6 +234,42 @@ unsigned int mraa_get_platform_pin_count(uint8_t platform_offset);
...
+*
+* @param pin_name: GPIO Pin Name. Eg: IO0
+* @return int of MRAA index for gpio
+*/
+
+int mraa_gpio_lookup(const char* pin_name);
+
+/**
+* Get pin number, board must be initialised.
```
Comment (pylbert) [inline api/mraa/common.h]: Same here, I think this should be:
Get I2C index by name...

File: include/mraa_internal_types.h [hunk 7]
```diff
@@ -143,6 +152,10 @@ struct _i2c {
     uint8_t* mock_dev_data; /**< mock device data register block contents */
 #endif
     /*@}*/
+#ifdef PERIPHERALMAN
+    AI2cDevice *bi2c;
+    char bus_name[256];
```
Comment (pylbert) [inline include/mraa_internal_types.h]: Is this set anywhere?

Comment (spitfire88) [inline include/mraa_internal_types.h]: Its not set anywhere. I don't see a use for bus_name field. Should be removed.

**Pull request merged**